pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -16,76 +15,107 @@ func Validate(client *Client, ic *types.InstallConfig) error {
 	platformPath := field.NewPath("platform").Child("alibabacloud")
 	allErrs = append(allErrs, validatePlatform(client, ic, platformPath)...)
 
-	if ic.ControlPlane != nil && ic.ControlPlane.Platform.AlibabaCloud != nil {
-		allErrs = append(allErrs, validateMachinePool(client, ic, field.NewPath("controlPlane", "platform", "alibabacloud"), ic.ControlPlane.Platform.AlibabaCloud, ic.ControlPlane.Replicas)...)
+	allErrs = append(allErrs, validateControlPlaneMachinePool(client, ic)...)
+	allErrs = append(allErrs, validateComputeMachinePool(client, ic)...)
+
+	return allErrs.ToAggregate()
+}
+
+func validateControlPlaneMachinePool(client *Client, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var mpool mergedMachinePool
+	defaultPool := alibabacloudtypes.DefaultMasterMachinePoolPlatform()
+	mpool.setWithFieldPath(&defaultPool, field.NewPath("controlPlane", "platform", "alibabacloud"))
+	if ic.Platform.AlibabaCloud != nil {
+		mpool.setWithFieldPath(ic.Platform.AlibabaCloud.DefaultMachinePlatform, field.NewPath("platform", "alibabacloud", "defaultMachinePlatform"))
 	}
+	mpool.setWithFieldPath(ic.ControlPlane.Platform.AlibabaCloud, field.NewPath("controlPlane", "platform", "alibabacloud"))
+
+	allErrs = append(allErrs, validateMachinePool(client, ic, &mpool)...)
+	return allErrs
+}
+
+func validateComputeMachinePool(client *Client, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var mpool mergedMachinePool
```

Comment:
Same comment here about typically code style.
```suggestion
	mpool := mergedMachinePool{}
```